### PR TITLE
Phase2-hgx298 Add first algo to get x,y of cell from u,v in rotated wafers

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Add first algo to get x,y of cell from u,v in rotated wafers

#### PR validation:

Tested using the runTheMatrix test workflows. Validated using the test code

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Nothing special

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by rovere** [comment]
assign hgcal-dpg

**Comment by bsunanda** [comment]
@rovere @pfs @cseez We would like it to be integrated now so that we can take the next step

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: Geometry/HGCalCommonData/src/HGCalWafer.cc [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+}
+
+std::pair<double, double> HGCalWafer::HGCalWaferUV2XY2(int32_t u, int32_t v, int32_t placementIndex, int32_t type) {
+  if (type != 0)
+    type = 1;
+  double x(0), y(0);
+  if (placementIndex < HGCalWafer::WaferPlacementExtra) {
+    double x0 = (1.5 * (u - v) - 0.5) * R_[type];
+    double y0 = (u + v - 2 * N_[type] + 1) * r_[type];
+    const std::vector<double> fac1 = {1.0, 0.5, -0.5, -1.0, -0.5, 0.5};
```
Comment (rovere) [inline Geometry/HGCalCommonData/src/HGCalWafer.cc:77]: Ciao @bsunanda would it make sense to rename `fac{1,2}` to something that could somehow suggest that you are performing a rotation, possibly with `sin` and `cos` inside the name?

Comment (bsunanda) [reply Geometry/HGCalCommonData/src/HGCalWafer.cc:77]: I am changing the names to fcos and fsin

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: Geometry/HGCalCommonData/interface/HGCalWafer.h [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+  std::pair<double, double> HGCalWaferUV2XY1(int32_t u, int32_t v, int32_t placementIndex, int32_t type);
+  std::pair<double, double> HGCalWaferUV2XY2(int32_t u, int32_t v, int32_t placementIndex, int32_t type);
+  std::pair<int32_t, int32_t> HGCalWaferUV2Cell(int32_t u, int32_t v, int32_t placementIndex, int32_t type);
+  static int32_t HGCalWaferPlacementIndex(int32_t iz, int32_t fwdBack, int32_t orient);
+
+private:
+  const double sqrt3By2_ = (0.5 * std::sqrt(3.0));
+  int32_t N_[2];
+  double R_[2], r_[2];
```
Comment (cvuosalo) [inline Geometry/HGCalCommonData/interface/HGCalWafer.h:47]: These variable names go against the CMS naming [guidelines](http://cms-sw.github.io/cms_coding_rules.html):

2.12 Do not use single character names, except for loop indices.
2.16 Use clear and explanatory variable names.

Please change them to help the reader understand them and to make them easier to search for in the code. Also, the names should be camel-case, starting with lower case.

Comment (bsunanda) [reply Geometry/HGCalCommonData/interface/HGCalWafer.h]: Changed names of variables

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by cvuosalo** [comment]
+1

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by cvuosalo** [comment]
+1

**Comment by srimanob** [comment]
+Upgrade

This PR introduces the method to get x,y of cell from u,v in rotated wafers. No change is expected from this PR. PR test runs fine.

**Comment by bsunanda** [comment]
@rovere @pfs please approve this PR. Without this we cannot proceed for progress

**Comment by perrotta** [comment]
+1

**Pull request merged**